cli/src/lockfiles/javascript.rs

```diff
@@ -91,16 +91,24 @@ impl Parseable for YarnLock {
             let resolution = package
                 .get(&"resolution".into())
                 .and_then(YamlValue::as_str)
+                .filter(|s| !s.is_empty())
                 .ok_or_else(|| "Failed to parse resolution field in yarn lock file".to_owned())?;
 
             // Ignore workspace-local dependencies like project itself ("project@workspace:.").
-            if resolution.contains("@workspace:") {
+            if resolution[1..].contains("@workspace:") {
                 continue;
             }
 
-            let (name, _version) = resolution
-                .rsplit_once("@npm:")
-                .ok_or_else(|| "Failed to parse name in yarn lock file".to_owned())?;
+            let (name, _version) = if let Some(index) = resolution[1..].find("@patch:") {
+                // Extract npm version from patched dependencies.
+                resolution[index + 8..]
+                    .rsplit_once("@npm")
+                    .ok_or_else(|| "Failed to parse patch in yarn lock file".to_owned())?
+            } else {
+                resolution
+                    .rsplit_once("@npm")
```

Comment:
In the lockfile which caused the issue, the `:` after npm is encoded as `%3A`. I've decided to remove it from both `rsplit_once` instances since it shouldn't be necessary for matching and could potentially cause problems in both places if urlencoding is allowed here.